internal/loader/loader.go

```diff
@@ -219,6 +220,24 @@ func normalizePackage(opts *config.IndexOpts, pkg *packages.Package) *packages.P
 
 			pkg.Module.Version = "."
 		}
+	} else if module.IsPseudoVersion(pkg.Module.Version) {
+		// Check if the module version is a pseudo-version.
+		// If it is, we will grab just the sha from it
+		// Note: According to the go mod spec (https://go.dev/ref/mod#versions) we expect
+		// versions to always follow semantic versions (either tagged version or pseudo-version).
+		// Go tidy will ensure that the version is a valid semantic version
+		// and replace non-canonical versions with v0.0.0
```

Comment:
```suggestion
        // Unpublished versions of dependencies have pseudo-versions in go.mod.
        // When the dependency itself is indexed, only the revision will be used.
        // For correct cross-repo navigation to such dependencies, only use 
        // the revision from a pseudo-version.
        //
        // The revision can also have build metadata following a `+`. Drop that,
        // similar to official Go tooling. (https://go.dev/ref/mod#versions)
        //
        // > The build metadata suffix is ignored for the purpose of comparing versions
```

1. It's important to focus on the Why rather than the What. Based on the code, the reader can tell that we'll just grab the revision if the version is a pseudo-version. It's more valuable to explain _why_ we're doing that.
2. I don't think we're guaranteed that the `go.mod` file has already been normalized by `go mod tidy`, so the comment related to that is not correct. Noticed this in the [docs](https://go.dev/ref/mod):

    > Versions in go.mod files may be [canonical](https://go.dev/ref/mod#glos-canonical-version) or non-canonical.

Reply:
Thanks for the suggestion. And this is way better.